CMake/config/CompilerFlagsHelpers.cmake

```diff
@@ -123,7 +123,7 @@ foreach(COMPILER_LANGUAGE ${SUPPORTED_COMPILER_LANGUAGE_LIST})
 
 	## rest of the world
 	else()
-		set(CMAKE_${COMPILER_LANGUAGE}_WARNING_ALL "-Wall")
+		set(CMAKE_${COMPILER_LANGUAGE}_WARNING_ALL "")
```

Comment:
You should create a section dedicated to PGI compiler where you remove `-Wall` instead of removing the flag in the default case.
This can help: https://cmake.org/cmake/help/v3.0/variable/CMAKE_LANG_COMPILER_ID.html

Reply:
That was my original approach. If you read above it's the opposite of what @pramodk told me to do. Which approach should I follow?

Reply:
I can't see the first patch because branch has been forced pushed.
But I would create a `elseif(compiler_is_pig)` section line 124 before the final `else`.

Reply:
> I can't see the first patch because branch has been forced pushed.
I didnt understand how GitHub allowed my local changes to get push forced....I think I fixed settings yesterday.

@tristan0x  is right.. I was thinking about cray and other compilers where -Wall may be right...but as Tristan suggested, you can unset Wall for PGI for now. See few lines below where we have done same for Intel.

Reply:
Pushed new version with separate PGI block after Tristan and Pramod comments.